framework/src/service/ServiceReferenceBasePrivate.cpp

```diff
@@ -119,41 +122,83 @@ std::shared_ptr<void> ServiceReferenceBasePrivate::GetService(BundlePrivate* bun
 
 InterfaceMapConstPtr ServiceReferenceBasePrivate::GetServiceInterfaceMap(BundlePrivate* bundle)
 {
+  /*
```

Comment:
Yeh, I have been thinking about this use case and could not come up with any strategy to detect recursion when multiple threads are involved. It's an edge case and users most likely won't do that. I am OK with this implementation.